src/cli/commands/install.js

```diff
@@ -494,7 +494,9 @@ export class Install {
     }
 
     // fin!
-    await this.saveLockfileAndIntegrity(topLevelPatterns, workspaceLayout);
+    if (topLevelPatterns.length || await fs.exists(path.join(this.config.cwd, constants.LOCKFILE_FILENAME))) {
+      await this.saveLockfileAndIntegrity(topLevelPatterns, workspaceLayout);
+    }
```

Comment:
I see, good thinking.
Yes, makes sense in that case.
Would you add a comment for that?